packages/react-strict-dom/src/native/modules/createStrictDOMComponent.js

```diff
@@ -679,31 +680,56 @@ export function createStrictDOMComponent<T, P: StrictProps>(
         );
       }
 
-      return (
-        <ThemeContext.Provider
-          value={
-            customPropertiesFromThemes != null
-              ? {
-                  ...inheritedCustomProperties,
-                  ...customPropertiesFromThemes
-                }
-              : inheritedCustomProperties
-          }
-        >
-          <DisplayModeInsideContext.Provider value={nextDisplayModeInside}>
-            <InheritableStyleContext.Provider
-              value={flattenStyle([
-                inheritedStyles as ?Style,
-                textStyles as ?Style
-              ])}
-            >
-              <FontSizeContext.Provider value={fontSizeValue}>
-                {element}
-              </FontSizeContext.Provider>
-            </InheritableStyleContext.Provider>
-          </DisplayModeInsideContext.Provider>
-        </ThemeContext.Provider>
-      );
+      if (fontSizeValue != null) {
+        element = (
+          <CSSFontSizeContext.Provider
+            children={element}
+            value={fontSizeValue}
+          />
+        );
+      }
+
+      if (
+        (inheritedStyles != null &&
+          typeof inheritedStyles === 'object' &&
+          Object.keys(inheritedStyles).length > 0) ||
+        (textStyles != null &&
+          typeof textStyles === 'object' &&
+          Object.keys(textStyles).length > 0)
+      ) {
+        element = (
```

Comment:
I think this can be optimized further (or I could be reading this wrong):

* if there are inherited styles but no text styles wouldn't that mean you can skip the provider?
* if there are text styles but no inherited styles you could set the provider value just to textStyles, skipping the flattening
* in the case where there are both you can keep it to what it's like here already

Reply:
Good point. Adding these scenarios to the unit test / app examples also surfaced that text inheritance is buggy, so I'm looking into that too.